music21/environment.py

```diff
@@ -484,33 +485,20 @@ def toSettingsXML(self, ref=None):
     def getDefaultRootTempDir(self):
         # noinspection SpellCheckingInspection
         '''
-        returns whatever tempfile.gettempdir() returns plus 'music21'.
+        returns self.defaultRootTempDir.
 
-        Creates the subdirectory if it doesn't exist:
-
-        >>> import tempfile
-        >>> import pathlib
-        >>> t = tempfile.gettempdir()
-        >>> #_DOCS_SHOW t
-        '/var/folders/x5/rymq2tx16lqbpytwb1n_cc4c0000gn/T'
+        if self.defaultRootTempDir isn't initialized, set it to
+        tempfile.TemporaryDirectory(prefix="music21-"), and it will
+        be created by system automatically.
 
         >>> import os
         >>> e = environment.Environment()
-        >>> e.getDefaultRootTempDir() == pathlib.Path(t) / 'music21'
+        >>> os.path.exists(e.getDefaultRootTempDir())
         True
         '''
-        # this returns the root temp dir; this does not create a new dir
-        dstDir = pathlib.Path(tempfile.gettempdir()) / 'music21'
-        # if this path already exists, we have nothing more to do
-        if dstDir.exists():
-            return dstDir
-        else:
-            # make this directory as a temp directory
-            try:
-                dstDir.mkdir()
-            except OSError:  # cannot make the directory
-                dstDir = dstDir.parent
-            return dstDir
+        if self.defaultRootTempDir is None:
+            self.defaultRootTempDir = tempfile.TemporaryDirectory(prefix="music21-")
+        return pathlib.Path(self.defaultRootTempDir.name)
```

Comment:
This is where it seems to me that User B would get the directory User A set for the entire environment.

Reply:
I think I'm good now. The singleton is set when the module is imported. Every Python user is going to need to import music21. Sorry for the noise!